src/storage/txn/commands/check_txn_status.rs

```diff
@@ -874,10 +959,113 @@ pub mod tests {
             ts(320, 0),
             r,
             false,
+            false,
             committed(ts(315, 0)),
         );
     }
 
+    #[test]
+    fn test_check_txn_status_resolving_pessimistic_lock() {
+        let engine = TestEngineBuilder::new().build().unwrap();
+        let k = b"k1";
+        let v = b"v1";
+        let ts = TimeStamp::compose;
+
+        // Check with resolving_pessimistic_lock flag.
+        // Path: there is no commit or rollback record, no rollback record should be written.
+        must_success(
+            &engine,
+            k,
+            ts(3, 0),
+            ts(3, 0),
+            ts(4, 0),
+            true,
+            false,
+            true,
+            |s| s == LockNotExistDoNothing,
+        );
+        must_get_none(&engine, k, ts(5, 0));
```

Comment:
`must_get_none` only checks there is no committed value. You can use `must_seek_write_none`.

Reply:
And same for below

Reply:
ah there's an existing comment now